Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionSidebarCocoa.mm

```diff
@@ -0,0 +1,220 @@
+}
+
+RetainPtr<CocoaImage> WebExtensionSidebar::icon(CGSize size)
+{
+    if (!extensionContext())
+        return nil;
+
+    const auto largestDim = [](CGSize size) {
+        return size.width > size.height ? size.width : size.height;
+    };
+
+    auto& context = extensionContext().value().get();
+    return m_iconsOverride
+        .and_then([&](RetainPtr<NSDictionary> icons) -> std::optional<RetainPtr<CocoaImage>> {
+            return toOptional(context.extension().bestImageInIconsDictionary(icons.get(), largestDim(size)));
+        })
+        .or_else([&] -> std::optional<RetainPtr<CocoaImage>> {
+            return parent().transform([&](auto const& parent) { return parent.get().icon(size); });
+        })
+        .value_or(context.extension().actionIcon(size));
+}
+
+void WebExtensionSidebar::setIconsDictionary(NSDictionary *icons)
+{
+    if (!icons || !icons.count) {
+        m_iconsOverride = std::nullopt;
+        return;
+    }
+
+    if (m_iconsOverride && [m_iconsOverride.value() isEqualToDictionary:icons])
+        return;
+
+    m_iconsOverride = icons;
+    propertiesDidChange();
+}
+
+String WebExtensionSidebar::title() const
+{
+    return m_titleOverride.value_or(
+        parent().transform([](auto const& parent) { return parent.get().title(); }).value_or(fallbackTitle)
+    );
+}
+
+void WebExtensionSidebar::setTitle(std::optional<String> titleOverride)
+{
+    m_titleOverride = titleOverride;
+    propertiesDidChange();
+}
+
+bool WebExtensionSidebar::isEnabled() const
+{
+    return m_isEnabled;
+}
+
+void WebExtensionSidebar::setEnabled(bool enabled)
+{
+    m_isEnabled = enabled;
+    propertiesDidChange();
+}
+
+bool WebExtensionSidebar::canProgrammaticallyOpenSidebar() const
+{
+    return extensionContext().transform([](auto const& context) -> bool { return !!context.get().extensionController(); })
+        .value_or(false);
+
+    // FIXME: <https://webkit.org/b/277575> also check that the controller delegate responds to whatever selector we use for this
+}
+
+void WebExtensionSidebar::openSidebarWhenReady()
+{
+    // FIXME: <https://webkit.org/b/277575> implement openSidebarWhenReady
+}
+
+bool WebExtensionSidebar::canProgrammaticallyCloseSidebar() const
+{
+    return extensionContext().transform([](auto const& context) -> bool { return !!context.get().extensionController(); })
+        .value_or(false);
+
+    // FIXME: <https://webkit.org/b/277575> also check that the controller delegate responds to whatever selector we use for this
+}
+
+void WebExtensionSidebar::closeSidebarWhenReady()
+{
+    // FIXME: <https://webkit.org/b/277575> implement closeSidebarWhenReady
+}
+
+String WebExtensionSidebar::sidebarPath() const
+{
+    String path = m_sidebarPathOverride.value_or(
+        parent().transform([](auto const& parent) { return parent.get().sidebarPath(); }).value_or(fallbackPath)
+    );
+    return path;
```

Comment:
```suggestion
    return m_sidebarPathOverride.value_or(
        parent().transform([](auto const& parent) { return parent.get().sidebarPath(); }).value_or(fallbackPath)
    );
```

Reply:
Whoops, this is left over from when I was resolving some typing issues this morning -- meant to remove it before pushing